src/sources.js

```diff
@@ -191,31 +191,35 @@ class CommunitySource extends Source {
     if (!repoName) throw new Error(`Cannot construct a ${this.constructor.name} without a repoName`);
 
     this.owner = owner;
-    [this.repoName, this.branch] = repoName.split(/@/, 2);
+    let urlDefinedBranch;
+    [this.repoName, urlDefinedBranch] = repoName.split(/@/, 2);
 
     if (!this.repoName) throw new Error(`Cannot construct a ${this.constructor.name} without a repoName after splitting on /@/`);
-    if (!this.branch) {
-      this.branch = fetch(`https://api.github.com/repos/${this.owner}/${this.repoName}`)
-        .then((res) => res.json())
-        .then((data) => data.default_branch || "master")
-        .catch(() => {
-          console.log(`Error interpreting the default branch of ${this.constructor.name}`);
-          return "master";
-        });
-    }
+
+    this.branch = fetch(`https://api.github.com/repos/${this.owner}/${this.repoName}`)
+      .then((res) => res.json())
+      .then((data) => ({
+        default: data.default_branch,
+        name: urlDefinedBranch || data.default_branch
+      }))
```

Comment:
Rather than modify the type of `this.branch` from a string to an object containing two values, I think the internal API might be simpler by keeping `this.branch` as the branch from the URL and adding `this.defaultBranch` as a new property. Thoughts?

Reply:
I see advantages to both (encapsulating branch-related information into `this.branch`, which I chose, or using properties such as `this.branch`, `this.defaultBranch`).

The only way I could achieve the latter is via code the following code, which I don't think is that nice. Do you have a better way?

```js
    let branchResolve, defaultBranchResolve;
    this.branch = new Promise((resolve) => {
      branchResolve = resolve;
    });
    this.branch = new Promise((resolve) => {
      defaultBranchResolve = resolve;
    });
    fetch(`https://api.github.com/repos/${this.owner}/${this.repoName}`)
      .then((res) => res.json())
      .then((data) => {
        branchResolve(urlDefinedBranch || data.default_branch);
        defaultBranchResolve(data.default_branch);
      })
      .catch(() => {
        console.log(`Error interpreting the default branch of ${this.constructor.name}`);
        branchResolve("master");
        defaultBranchResolve("master");
      });
```

Reply:
> Do you have a better way?

I believe something like the below would work just fine. Diff shown against the clean slate of `master` to more clearly demonstrate the changes:

```diff
--- a/src/sources.js
+++ b/src/sources.js
@@ -194,19 +194,31 @@ class CommunitySource extends Source {
     [this.repoName, this.branch] = repoName.split(/@/, 2);
 
     if (!this.repoName) throw new Error(`Cannot construct a ${this.constructor.name} without a repoName after splitting on /@/`);
+
+    this.defaultBranch = fetch(`https://api.github.com/repos/${this.owner}/${this.repoName}`)
+      .then((res) => res.json())
+      .then((data) => data.default_branch || "master")
+      .catch(() => {
+        console.log(`Error interpreting the default branch of ${this.constructor.name}`);
+        return "master";
+      });
+
     if (!this.branch) {
-      this.branch = "master";
+      this.branch = this.defaultBranch;
     }
   }
 
@@ -205,6 +216,7 @@ class CommunitySource extends Source {
 
-  get repoNameWithBranch() {
-    return this.branch === "master"
+  async repoNameWithBranch() {
+    const branch = await this.branch;
+    return branch === await this.defaultBranch
       ? this.repoName
-      : `${this.repoName}@${this.branch}`;
+      : `${this.repoName}@${branch}`;
   }
 
```

Reply:
Resolved in 
d81afdc